msibi/pair.py

```diff
@@ -47,32 +48,59 @@ def __init__(self, type1, type2, potential, head_correction_form="linear"):
         self.head_correction_form = head_correction_form
 
     def add_state(
-        self, state, target_rdf, alpha, pair_indices=None, alpha_form="linear"
+        self,
+        state,
+        target_rdf=None,
+        calculate_target_rdf=False,
+        pair_indices=None,
+        alpha_form="linear"
     ):
         """Add a state to be used in optimizing this pair.
 
         Parameters
         ----------
         state : State
             A state object.
-        target_rdf : np.ndarray, shape=(n_bins, 2), dtype=float
-            Coarse-grained target RDF.
-        alpha : float
-            The alpha value used to scale the weight of this state.
         pair_indices : array-like (n_pairs, 2) dtype=int
             Each row gives the indices of two atoms representing a pair
             (default None)
         alpha_form : str
             For alpha as a function of r, gives form of alpha function
             (default 'linear')
         """
+        if calculate_target_rdf and target_rdf != None:
+            raise ValueError(
+                    "Setting calcualte_target_rdf = True will overwirte "
+                    "the data passed to target_rdf. calculate_target_rdf "
+                    "should only be used when target_rdf is None"
+                    )
+        if target_rdf:
+            if os.path.isfile(target_rdf):
+                try:
+                    target_rdf = np.loadtxt(target_rdf)
+                except Exception as e:
+                    print(e)
+            elif isinstance(target_rdf, np.ndarray):
+                pass
+
+        elif calculate_target_rdf:
+            target_rdf = state_pair_rdf(state, self)
```

Comment:
We might want to make this logic a little more robust. Either the user provides an rdf or one is calculated. Unless is there another situation?
```suggestion
        else:
            target_rdf = state_pair_rdf(state, self)
```

Reply:
I guess in total there are 4 scenarios. Both are True, one or the other is given, or neither are given.  I added an additional else statement that catches the scenario where nothing is given to `target_rdf` and `calculate_target_rdf=False`